packages/root/src/Root.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { HTMLElementProps } from '@leafygreen-ui/lib';
+
+type ButtonProps<T> = HTMLElementProps<'button'> &
+  T & {
+    href?: undefined;
+  };
+
+type Anchor<T> = HTMLElementProps<'a'> &
+  T & {
+    href: string;
+  };
+
+type CustomElementProps<T> = T & {
+  as: React.ElementType<any>;
+  [key: string]: any;
+};
+
+type RootProps<T> = ButtonProps<T> | Anchor<T> | CustomElementProps<T>;
```

Comment:
We should probably export this type, right? So then when this gets reused by `Button` it can use it like...

```ts
type ButtonProps = RootProps<SharedButtonProps>
```

Reply:
Yes that makes so much sense... good call